pyjob/sge.py

```diff
@@ -104,13 +104,15 @@ def _create_runscript(self):
             cmd = '-q {}'.format(self.queue)
             runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
         if self.runtime:
-            cmd = '-l h_rt={}'.format(self.runtime)
+            h, m = divmod(self.runtime, 60)
+            m, s = divmod(m, 60)
+            cmd = '-l h_rt={0:02d}:{1:02d}:{2:02d}'.format(h, m, s)
             runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
         if self.shell:
             cmd = '-S {}'.format(self.shell)
             runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
         if self.nprocesses:
-            cmd = '-pe mpi {}'.format(self.nprocesses)
+            cmd = '-pe {} {}'.format(self.environment, self.nprocesses)
```

Comment:
what options are allowed? do we need to validate what we should allow for environment or trust the user to provide something reasonable

Reply:
The list of available environments depends on the specific configuration of the system, so it varies from cluster to cluster. A complete list of environments can be obtained using "qconf -spl". I guess it would be possible to create a function to run this command and only allow for environments listed with this command. This would actually solve a bug observed when running tasks through pyjob on a cluster with no specific environment setup (hpc02-ccp4 cluster, all cores allocated in the same node and no environment has been created). This cluster has no environments, and default "#$ -pe mpi 1" card crashes the task.